src/main/resources/archetype-resources/pom.xml

```diff
@@ -0,0 +1,269 @@
+    <elemental2.version>1.0.0</elemental2.version>
+
+    <google.jsinterop.version>1.0.0</google.jsinterop.version>
+    <j2cl.version>${j2cl.version}</j2cl.version>
+
+    <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
+    <maven.compiler.source>${maven.compiler.target}</maven.compiler.source>
+
+    <maven.compiler.plugin.version>3.8.0</maven.compiler.plugin.version>
+    <maven.tomcat7.plugin.version>2.2</maven.tomcat7.plugin.version>
+    <maven.jetty.plugin.version>9.4.26.v20200117</maven.jetty.plugin.version>
+    <maven.source.plugin.version>${maven.source.plugin.version}</maven.source.plugin.version>
+    <maven.resources.plugin.version>3.1.0</maven.resources.plugin.version>
+    <maven.war.plugin.version>3.2.3</maven.war.plugin.version>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.elemental2</groupId>
+      <artifactId>elemental2-dom</artifactId>
+      <version>${elemental2.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.jsinterop</groupId>
+      <artifactId>base</artifactId>
+      <version>${google.jsinterop.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.vertispan.j2cl</groupId>
+      <artifactId>junit-annotations</artifactId>
+      <version>\${j2cl.version}</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>com.vertispan.j2cl</groupId>
+      <artifactId>gwttestcase-emul</artifactId>
+      <version>\${j2cl.version}</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>com.vertispan.j2cl</groupId>
+      <artifactId>gwttestcase-emul</artifactId>
+      <version>\${j2cl.version}</version>
+      <classifier>sources</classifier>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>com.vertispan.j2cl</groupId>
+      <artifactId>junit-emul</artifactId>
+      <version>\${j2cl.version}</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>com.vertispan.j2cl</groupId>
+      <artifactId>junit-emul</artifactId>
+      <version>\${j2cl.version}</version>
+      <classifier>sources</classifier>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+        <version>${maven.resources.plugin.version}</version>
+        <executions>
+          <execution>
+            <id>custom-resources</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>\${project.build.directory}</outputDirectory>
+              <resources>
+                <resource>
+                  <targetPath>\${basedir}/target/gwt/launcherDir/\${artifactId}</targetPath>
```

Comment:
curious as to why you did this instead of making j2cl-m-p use the war dir?

Reply:
because j2cl-m-p generates js and sources under artifact name like demo/demo.js and demo/sources, in this case sources can't be located, so i have to use demo as a root dir.